lib/events.js

```diff
@@ -279,18 +276,14 @@ EventEmitter.prototype.removeListener =
       if (!events)
         return this;
 
-      list = events[type];
+      list = events.get(type);
       if (!list)
         return this;
 
       if (list === listener || (list.listener && list.listener === listener)) {
-        if (--this._eventsCount === 0)
-          this._events = {};
-        else {
-          delete events[type];
-          if (events.removeListener)
-            this.emit('removeListener', type, listener);
-        }
+        events.delete(type);
```

Comment:
There is no significant performance difference if I revert ca7507253a1d32e2c54452b22ec020af7e6225a4. I tested with `this._events.clear()` and `this._events = new Map()`